contribute/design/1.x/apis/cstorvolume_policy.md

```diff
@@ -0,0 +1,318 @@
+# Move `CStorVolumeConfig`, `CStorVolumePolicy` resources Schema to version `v1`
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Goals](#goals)
+* [Non-Goals](#non-goals)
+* [VolumeConfig Schema Proposal](#volume-config-schema-proposal)
+* [VolumePolicy Schema Proposal](#volume-policy-schema-proposal)
+* [Risk and Mitigations](#risks-and-mitigations)
+* [CRD Migration and Graduation](#graduation-criteria)
+
+## Introduction
+
+This proposal highlights the limitations of current `CStorVolumeConfig` and `CStorVolumePolicy` schema and
+proposes improvements and feature changes.
+
+# Goals
+
+The major goal of this document is to freeze the APIs schema for `CStorVolumeConfig` and `CStorVolumePolicy`.
+Apart from this the document focuses on following aspects too:
+
+- Migrating the `CStorVolumeConfig` and `CStorVolumePolicy` CRD to `v1` apiversion inder `cstor.openebs.io` api group.
+- Identify the breaking and upgrade changes and documentation around that.
+- List of possible high level status fields on `CStorVolumeConfig` and `CStorVolumePolicy`.
+
+# Non-Goals
+
+- Introduction of new fields in the `CStorVolumeConfig` and `CStorVolumePolicy` to deliver feature etc.
+- List of possible events that should be recorded on `CStorVolumeConfig` and `CStorVolumesPolicy`.
+- List of conditions that should be put on the `CStorVolumeConfig` and `CStorVolumePolicy` CRs.
+
+# Volume Config Schema Proposal
+
+## `CStorVolumeConfig` APIs
+
+```go
+
+// CStorVolumeConfig describes a cstor volume config resource created as
+// custom resource. CStorVolumeConfig is a request for creating cstor volume
+// related resources like target deployment, replica resources and service etc.
+type CStorVolumeConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec defines a specification of a cstor volume claim required
+	// to provisione cstor volume resources
+	Spec CStorVolumeConfigSpec `json:"spec"`
+
+	// Publish contains info related to attachment of a volume to a node.
+	// i.e. NodeId etc.
+	Publish CStorVolumeConfigPublish `json:"publish,omitempty"`
+
+	// Status represents the current information/status for the cstor volume
+	// claim, populated by the controller.
+	Status         CStorVolumeConfigStatus `json:"status"`
+  // VersionDetails contains version related info
+  VersionDetails VersionDetails         `json:"versionDetails"`
+}
+
+// CStorVolumeConfigSpec is the spec for a CStorVolumeConfig resource
+type CStorVolumeConfigSpec struct {
+	// Capacity represents the actual resources of the underlying
+	// cstor volume.
+	Capacity corev1.ResourceList `json:"capacity"`
```

Comment:
We required few other fields InitialCapacity and replicaCount which users can't modify after updating them.
```go
type InitConfig struct {
    ReplicaCount int `json:"replicaCount"`
    Capacity corev1.ResourceList `json:"capacity"`
}
```
```go
type CStorVolumeConfigSpec struct {
    InitialConfig InitConfig `json:"initConfig"`
}
```
InitialCapacity is required to create new CVR's as part of CVR ScaleUp.
cc: @vishnuitta Your thoughts on naming conventions.

Reply:
@vishnuitta can we get the initial capacity from the existing CVRs instead of getting it from CVC. as we sync the data why not sync the other details from the CVR resources.

Reply:
these two should be readOnly attributes.. they shouldn't get modified

Reply:
As discussed we will have it as `CStorVolumePolicy.Spec.Provision`